internal/validator/pipeline_test.go

```diff
@@ -0,0 +1,134 @@
+package validator
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	v1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// Verifies that matrix-supplied params satisfy required Task params and types
+// when the Task is embedded via taskSpec and PipelineTask.Params are omitted.
+func TestValidatePipeline_MatrixParamsSatisfyTaskParams(t *testing.T) {
+	// Prepare a temp directory and write a minimal Pipeline YAML that uses matrix params
+	tmpDir, err := os.MkdirTemp("", "pipeline-matrix-test")
+	if err != nil {
+		t.Fatalf("failed to create temp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	pipelineYAML := `apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: matrix-pipeline
+spec:
+  tasks:
+    - name: run
+      taskSpec:
+        params:
+          - name: img
+            type: string
+          - name: tags
+            type: array
+        steps:
+          - name: s
+            image: "$(params.img)"
+            script: |
+              echo processing
+      matrix:
+        params:
+          - name: img
+            value:
+              - alpine:3.18
+              - alpine:3.19
```

Comment:
What if, for example, one of the items in this list was an array:

```yaml
value:
  - alpine:3.18
  - [some, odd, stuff]
  - alpine:3.19
```

I'd expect the validation to also fail. Can we add a test to ensure that's covered? This is different than what `TestValidatePipeline_MatrixPlatformParamPresent` is checking. I want us to make sure that we're catching when at least one of the values is invalid. Order may matter.

Reply:
One thing we could do is make sure that all the values are of the same type, then we can get away with only using the first value as the first value.